components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -543,18 +542,15 @@ impl CanvasRenderingContext2D {
                         Some(ref canvas) => &**canvas,
                     };
 
-                    let window = window_from_node(canvas);
-
-                    let style = window.GetComputedStyle(canvas.upcast(), None);
+                    let style = canvas.upcast::<Element>().style();
 
                     let element_not_rendered =
-                        !canvas.upcast::<Node>().is_in_doc() ||
-                        style.GetPropertyValue(DOMString::from("display")) == "none";
+                        !canvas.upcast::<Node>().is_in_doc() || style.is_none();
 
                     if element_not_rendered {
                         Ok(RGBA::new(0, 0, 0, 255))
                     } else {
-                        self.parse_color(&style.GetPropertyValue(DOMString::from("color")))
+                        Ok(style.unwrap().get_color().color)
                     }
```

Comment:
I'm wondering will it be better to use pattern matching for `style` here?

```rust
let style = canvas.upcast::<Element>().style();

match (style) {
  Some(s) if canvas.upcast::<Node>().is_in_doc() => Ok(s.get_color().color),
  _ => Ok(RGBA::new(0, 0, 0, 255))
}
```

Reply:
I was pondering on that as well but couldn't think of a good way to write it. Thanks for the suggestion, I'll try it out.